src/core/models/highlightedfieldmap.js

```diff
@@ -0,0 +1,76 @@
+import HighlightedValue from '../models/highlightedvalue';
+
+/**
+ * Represents mappings from fields to highlighted strings
+ */
+export default class HighlightedFieldMap {
+  /**
+   * Constructs a highlighted field map which consists of mappings from fields to highlighted
+   * value strings.
+   *
+   * @param {Object<string, string|object>} data Keyed by fieldName. It may consist of nested fields
+   *
+   * Example object:
+   *
+   * {
+   *   name: '<strong>Yext</strong>',
+   *   description: {
+   *     featured: '<strong>Yext</strong> is the offical answers company'
```

Comment:
Is this the same structure as the object output by `ResultFactory.computeHighlightedData`?

Reply:
yes